astropy/cosmology/parameter/_converter.py

```diff
@@ -38,12 +58,12 @@ def _register_validator(key, fvalidate=None):
         return fvalidate
 
     # for use as a decorator
-    def register(fvalidate):
+    def register(fvalidate: _FValidateCallable) -> _FValidateCallable:
         """Register validator function.
 
         Parameters
         ----------
-        fvalidate : callable[[object, object, Any], Any]
+        fvalidate : callable[[Cosmology, Parameter, Any], Any]
```

Comment:
Same. There seems to be no instruction, so I'm following `typing`. Hopefully we can eventually drop type info in the docs in favor of the type annotation.

Reply:
This is going beyond the scope of this pull request, but do you have any experience with [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints)?

Reply:
Yes, some.
On another out-of-scope tangent. https://github.com/tox-dev/sphinx-autodoc-typehints/issues/407 wold be lovely.
Then for parameters that don't require long explanations we could embed basically everything directly in the function signature. IDE's and coding AIs would be OP if they could directly accessing docstrings without having to worry about format.

Reply:
At the coordination meeting the question of duplicating parameter types in annotations and in docstrings was brought up, so it would be good to think about how that duplication could be avoided.

Reply:
Agreed. Getting [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints) and numpydoc to play well together might be tricky. But removing all redundant type info from the docs would be excellent.